react/components/contexts/auth-context/auth-context.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from "react";
+import AsyncStorage from "@react-native-community/async-storage";
+import PropTypes from "prop-types";
+import { verify } from "yonius";
+import { API as RipeIdAPI } from "ripe-id-api";
+
+import { RipeContext } from "../ripe-context";
+
+export const AuthContext = React.createContext();
+
+export class AuthProvider extends Component {
+    static contextType = RipeContext;
+
+    static get propTypes() {
+        return {
+            options: PropTypes.shape({
+                clientId: PropTypes.string.isRequired,
+                clientSecret: PropTypes.string.isRequired,
+                redirectUrl: PropTypes.string.isRequired
+            })
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.ripeIdApi = new RipeIdAPI(this.props.options);
+        this.state = {
+            loaded: false,
+            logoutMessage: null,
+            account: null,
+            tokens: null,
+            ripeIdApi: this.ripeIdApi
+        };
+    }
+
+    async trySetAccount() {
+        const sid = await AsyncStorage.getItem("sessionId");
+        const account = await AsyncStorage.getItem("account");
+        const tokens = await AsyncStorage.getItem("tokens");
+
+        if (!sid || !account || !tokens) return false;
+
+        this.ripeIdApi.sessionId = sid;
+        this.setState({
+            account: account ? JSON.parse(account) : null,
+            tokens: tokens ? JSON.parse(tokens) : null
+        });
+        return true;
+    }
+
+    async login(email, password) {
+        const response = await this.ripeIdApi.login(email, password);
+        verify(response.sid, "No session ID found", 404);
+        verify(response.tokens, "No tokens list found", 404);
+
+        this.ripeIdApi.sessionId = response.sid;
+
+        const account = await this.ripeIdApi.selfAccount();
+        verify(account, "No account found", 404);
+
+        await AsyncStorage.setItem("sessionId", response.sid);
+        await AsyncStorage.setItem("account", JSON.stringify(account));
+        await AsyncStorage.setItem("tokens", JSON.stringify(response.tokens));
+        await this.trySetAccount();
+    }
+
+    async logout(message = null) {
+        // re-creates the RIPE ID API instance and invalidates the
+        // current RIPE SDK instance as it's no longer going to be used
+        this.ripeIdApi = new RipeIdAPI(this.props.options);
+        if (this.context.ripeApi) this.context.ripeApi.unauth();

```

Comment:
Is this valid in this context now?

Reply:
Nope :+1:

Reply:
Does it also need to recreate the ripe ID?

Reply:
Think so, if you logout you basically want to erase every token, the approach on  the oauth context is to recreate the api object (clears session and auth tokens)

Reply:
Yeah, you are right